dotnet/src/Agents/OpenAI/OpenAIAssistantAgent.cs

```diff
@@ -237,7 +230,7 @@ public async Task<bool> DeleteThreadAsync(
         // Validate input
         Verify.NotNullOrWhiteSpace(threadId, nameof(threadId));
 
-        return await this._client.DeleteThreadAsync(threadId, cancellationToken).ConfigureAwait(false);
+        return (await this._client.DeleteThreadAsync(threadId, cancellationToken).ConfigureAwait(false)).Value.Deleted;
```

Comment:
This may be a little more clear to maintainers and with less _cruft_:

```suggestion
        ThreadDeletionResult result = await this._client.DeleteThreadAsync(threadId, cancellationToken).ConfigureAwait(false);

        return result.Deleted;
```